homeassistant/components/google_domains.py

```diff
@@ -0,0 +1,76 @@
+"""Integrate with Google Domains."""
+import asyncio
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_DOMAIN, CONF_PASSWORD, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+DOMAIN = 'google_domains'
+UPDATE_URL = 'https://{}:{}@domains.google.com/nic/update'
+INTERVAL = timedelta(minutes=5)
+_LOGGER = logging.getLogger(__name__)
+SERVICE_UPDATE_DNS = 'update_dns'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_DOMAIN): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Initialize the Google Domains component."""
+    domain = config[DOMAIN][CONF_DOMAIN]
+    user = config[DOMAIN][CONF_USERNAME]
+    password = config[DOMAIN][CONF_PASSWORD]
+    session = async_get_clientsession(hass)
+
+    result = yield from _update_google_domains(session, domain, user, password)
+
+    if not result:
+        return False
+
+    @asyncio.coroutine
+    def update_domain_interval(now):
+        """Update the Google Domains entry."""
+        yield from _update_google_domains(session, domain, user, password)
+
+    @asyncio.coroutine
+    def update_domain_service(call):
+        """Update the Google Domains entry."""
+        yield from _update_google_domains(session, domain, user, password)
+
+    async_track_time_interval(hass, update_domain_interval, INTERVAL)
+    hass.services.async_register(
```

Comment:
This service was removed from the namecheap DNS in #9845 but I don't completely understand what the harm is in having a service that you can call to manually trigger an update. Do I need to remove this from here as well?